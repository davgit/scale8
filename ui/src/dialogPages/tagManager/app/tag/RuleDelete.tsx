import { FC } from 'react';
import DeleteRuleQuery from '../../../../gql/mutations/DeleteRuleQuery';
import { DeleteRule } from '../../../../gql/generated/DeleteRule';
import { DialogPageProps } from '../../../../types/DialogTypes';
import {
    DialogDeletePreview,
    DialogDeletePreviewProps,
} from '../../../abstractions/DialogDeletePreview';
import { DialogMutationFunction } from '../../../abstractions/DialogDirectMutation';
import { FormMutationData } from '../../../../hooks/form/useFormValidation';
import { DeleteActionGroup } from '../../../../gql/generated/DeleteActionGroup';
import { modelsDeleteReducer } from '../../../../utils/ArrayUtils';

const RuleDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    const buildModelsDeletedMap = (data: DeleteRule) =>
        data.deleteRule.reduce(
            modelsDeleteReducer,
            new Map<string, { name: string; id: string }[]>(),
        );

    const findName = (data: FormMutationData | DeleteActionGroup) => {
        const modelsDeletedMap = buildModelsDeletedMap(data as DeleteRule);

        const ruleModels = modelsDeletedMap.get('Rule');
        return ruleModels ? ruleModels[0].name : '';
    };

    const dialogDeletePreviewProps: DialogDeletePreviewProps = {
        mutation: DeleteRuleQuery,
        executePreviewMutationCallback: async (
            mutationFunction: DialogMutationFunction,
            id: string,
        ) => {
            await mutationFunction({
                variables: {
                    ruleDeleteInput: {
                        rule_id: id,
                        preview: true,
                    },
                },
            });
        },
        executeMutationCallback: async (
            mutationFunction: DialogMutationFunction,
            id: string,
            contextId: string,
            ids: string[],
            comments: string,
        ) => {
            await mutationFunction({
                variables: {
                    ruleDeleteInput: {
                        rule_id: id,
                        ...(comments === '' ? {} : { comments }),
                    },
                },
            });
        },
        buildText: (data: FormMutationData | DeleteRule) => {
            return `Deleting the rule "${findName(data)}" will also delete:`;
        },
        buildTextNoGroups: (data: FormMutationData | DeleteActionGroup) => {
            return `Delete the rule: ${findName(data)}?`;
        },
        buildGroups: (data: FormMutationData | DeleteRule) => {
            const modelsDeletedMap = buildModelsDeletedMap(data as DeleteRule);

            const groups: {
                title: string;
                items: { name: string; id: string }[];
            }[] = [];

            [
                { model: 'Event', label: 'Events' },
                { model: 'ConditionRule', label: 'Condition Rules' },
                { model: 'ExceptionRule', label: 'Exception Rules' },
                {
                    model: 'ActionGroupDistribution',
                    label: 'Action Group Distributions',
                },
                { model: 'ActionGroup', label: 'Action Groups' },
                { model: 'Action', label: 'Actions' },
                { model: 'DataMap', label: 'Data Maps' },
                { model: 'RepeatedDataMap', label: 'Repeated Data Maps' },
            ].forEach((_) => {
                const items = modelsDeletedMap.get(_.model);
                if (items !== undefined) {
                    groups.push({
                        title: `${_.label} (${items.length})`,
                        items,
                    });
                }
            });

            return groups;
        },
        ...props,
    };

    return <DialogDeletePreview {...dialogDeletePreviewProps} />;
};

export default RuleDelete;
