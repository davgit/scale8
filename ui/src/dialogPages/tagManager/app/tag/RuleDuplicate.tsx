import { FC } from 'react';
import { FormValidationResult } from '../../../../hooks/form/useFormValidation';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import DuplicateRuleQuery from '../../../../gql/mutations/DuplicateRuleQuery';
import DuplicateDialogForm, {
    DuplicateDialogProps,
} from '../../../../components/organisms/Forms/DuplicateDialogForm';
import DuplicateRuleGetQuery from '../../../../gql/queries/DuplicateRuleGetQuery';
import { DuplicateRuleGetData } from '../../../../gql/generated/DuplicateRuleGetData';
import { DuplicateRule } from '../../../../gql/generated/DuplicateRule';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../../abstractions/DialogPreloadForm';
import {
    DuplicateValidators,
    DuplicateValues,
} from '../../../../utils/forms/DuplicateDialogFormUtils';
import { generateRevisionName } from '../../../../../../common/utils/GenerateRevisionName';

const RuleDuplicate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const duplicateProps: DialogPreloadFormProps<
        DuplicateRuleGetData,
        DuplicateValues,
        DuplicateDialogProps,
        DuplicateRule
    > = {
        loadQuery: useQuery<DuplicateRuleGetData>(DuplicateRuleGetQuery, {
            variables: { id: props.id },
        }),
        buildInitialStatePreload: () => ({
            name: generateRevisionName(),
        }),
        saveQuery: useMutation<DuplicateRule>(DuplicateRuleQuery),
        mapSaveData: (formValues: DuplicateValues) => ({
            ruleDuplicateInput: {
                name: formValues.name,
                rule_id: props.id,
            },
        }),
        buildFormProps: (
            formLoadedData: DuplicateRuleGetData,
            formValidationValues: FormValidationResult<DuplicateValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Create Rule',
            title: 'Create Rule',
            handleDialogClose: props.handleDialogClose,
            oldName: formLoadedData.getRule.name,
            description: `Create a new Rule from ${formLoadedData.getRule.name}?`,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.duplicateRule.id !== undefined,
        pageComponent: DuplicateDialogForm,
        validators: DuplicateValidators,
        ...props,
    };

    return (
        <DialogPreloadForm<
            DuplicateRuleGetData,
            DuplicateValues,
            DuplicateDialogProps,
            DuplicateRule
        >
            {...duplicateProps}
        />
    );
};

export default RuleDuplicate;
