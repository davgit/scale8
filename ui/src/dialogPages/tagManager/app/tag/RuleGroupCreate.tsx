import { FC } from 'react';
import RuleGroupForm from '../../../../components/organisms/Forms/RuleGroupForm';
import { FormProps, FormValidationResult } from '../../../../hooks/form/useFormValidation';
import { ApolloError, useMutation } from '@apollo/client';
import CreateRuleGroupQuery from '../../../../gql/mutations/CreateRuleGroupQuery';
import nameValidator from '../../../../utils/validators/nameValidator';
import { CreateRuleGroup } from '../../../../gql/generated/CreateRuleGroup';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { DialogForm, DialogFormProps } from '../../../abstractions/DialogForm';

export type RuleGroupValues = {
    name: string;
    comments: string;
};

export type RuleGroupFormProps = FormProps<RuleGroupValues>;

const RuleGroupCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const tagId = props.contextId;

    const ruleGroupCreateProps: DialogFormProps<
        RuleGroupValues,
        RuleGroupFormProps,
        CreateRuleGroup
    > = {
        buildInitialState: () => ({
            name: '',
            comments: '',
        }),
        saveQuery: useMutation(CreateRuleGroupQuery),
        mapSaveData: (ruleGroupValues: RuleGroupValues) => ({
            ruleGroupCreateInput: {
                tag_id: tagId,
                name: ruleGroupValues.name,
                ...(ruleGroupValues.comments === ''
                    ? {}
                    : {
                          comments: ruleGroupValues.comments,
                      }),
            },
        }),
        buildFormProps: (
            formValidationValues: FormValidationResult<RuleGroupValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Create Rule Group',
            title: 'Create Rule Group',
            formInfoProps: buildStandardFormInfo('ruleGroups', 'Create'),
            handleDialogClose: props.handleDialogClose,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createRuleGroup.id !== undefined,
        pageComponent: RuleGroupForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Rule Group name too short',
            },
        ],
        ...props,
    };

    return (
        <DialogForm<RuleGroupValues, RuleGroupFormProps, CreateRuleGroup>
            {...ruleGroupCreateProps}
        />
    );
};

export default RuleGroupCreate;
