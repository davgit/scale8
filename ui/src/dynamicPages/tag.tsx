import { FC, Fragment } from 'react';
import { Box } from '@mui/material';
import { useQuery } from '@apollo/client';
import { DynamicPageProps } from '../pageLoader/DynamicPageLoader';
import { useLoggedInState } from '../context/AppContext';
import { QueryLoaderAndError } from '../abstractions/QueryLoaderAndError';
import { TagContentPageData } from '../gql/generated/TagContentPageData';
import PageTagContentQuery from '../gql/queries/PageTagContentQuery';
import { PageActionProps, pageActions } from '../actions/PageActions';
import { useRouter } from 'next/router';
import { toAppRevision } from '../utils/NavigationPaths';
import NonTablePageContainer from '../components/molecules/NonTablePageContainer';
import { buildStandardMainInfo } from '../utils/InfoLabelsUtils';
import RuleContainerDivider from '../components/atoms/RuleContainerDivider';
import { RuleGroupPageSection } from '../components/organisms/Sections/RuleGroupPageSection';
import { RuleGroup } from '../types/TagRulesTypes';
import { AlertRevisionFinal } from '../components/atoms/AlertRevisionFinal';
import { SmallAddButton } from '../components/atoms/SmallAddButton';

const RulesPage: FC<DynamicPageProps> = (props: DynamicPageProps) => {
    const tagId = props.params.id ?? '';

    const router = useRouter();

    const { templateInteractions } = useLoggedInState();
    const { dispatchDialogAction } = templateInteractions;

    return QueryLoaderAndError<TagContentPageData>(
        true,
        useQuery(PageTagContentQuery, {
            variables: { id: tagId },
        }),
        (data: TagContentPageData, valuesRefresh: (mustResetCache: boolean) => void) => {
            const pageActionProps: PageActionProps = {
                dispatchDialogAction,
                refresh: (mustResetTable: boolean, mustResetCache: boolean) => {
                    valuesRefresh(mustResetCache);
                },
            };

            return (
                <Box>
                    {data.getTag.revision.locked && (
                        <AlertRevisionFinal
                            onCloneLinkClick={() => {
                                pageActions.duplicateAppRevision(
                                    pageActionProps,
                                    data.getTag.revision.id,
                                    (
                                        id: string,
                                        pageRefresh: () => void,
                                        handleDialogClose: (checkChanges: boolean) => void,
                                    ) => {
                                        handleDialogClose(false);
                                        router.push(toAppRevision({ id }, 'tags')).then();
                                    },
                                );
                            }}
                        />
                    )}
                    <NonTablePageContainer title="Rules" {...buildStandardMainInfo('rules')}>
                        {data.getTag.rule_groups.map((ruleGroup, ruleGroupIndex) => (
                            <Fragment key={ruleGroup.id}>
                                {ruleGroupIndex !== 0 && <RuleContainerDivider text="And" />}
                                <RuleGroupPageSection
                                    ruleGroup={ruleGroup as RuleGroup}
                                    ruleGroupIndex={ruleGroupIndex}
                                    ruleGroupsIds={data.getTag.rule_groups.map(
                                        (ruleGroup) => ruleGroup.id,
                                    )}
                                    tagId={tagId}
                                    valuesRefresh={valuesRefresh}
                                    revisionLocked={data.getTag.revision.locked}
                                    pageActionProps={pageActionProps}
                                />
                            </Fragment>
                        ))}
                        <SmallAddButton
                            addButtonText="Add Rule Group"
                            addButtonClick={() =>
                                pageActions.createRuleGroup(pageActionProps, tagId)
                            }
                            disabled={data.getTag.revision.locked}
                        />
                    </NonTablePageContainer>
                </Box>
            );
        },
        true,
    );
};

export default RulesPage;
