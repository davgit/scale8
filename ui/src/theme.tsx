import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
