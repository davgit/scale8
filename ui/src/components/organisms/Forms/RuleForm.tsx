import { FC, useEffect, useState } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import { Checkbox, FormControlLabel } from '@mui/material';
import ControlledTextAreaInput from '../../atoms/ControlledInputs/ControlledTextAreaInput';
import PositiveInRangeInput from '../../atoms/InputTypes/PositiveInRangeInput';
import { RuleFormProps } from '../../../dialogPages/tagManager/app/tag/RuleCreate';

const RuleForm: FC<RuleFormProps> = (props: RuleFormProps) => {
    const { values, availableGlobalTriggers, handleChange } = props;

    const [useGlobalTrigger, setUseGlobalTrigger] = useState(false);

    useEffect(() => {
        if (!useGlobalTrigger && values.globalTriggerId !== '') {
            handleChange('globalTriggerId', '');
        }
    }, [values, useGlobalTrigger, handleChange]);

    return (
        <DrawerFormLayout
            submitDisable={
                props.isSubmitting ||
                (availableGlobalTriggers !== undefined &&
                    useGlobalTrigger &&
                    availableGlobalTriggers.length === 0)
            }
            {...props}
        >
            <ControlledTextInput
                name="name"
                label="Name"
                formProps={props}
                className="DrawerFormField"
                required
                autoFocus
            />
            {availableGlobalTriggers !== undefined && (
                <>
                    <FormControlLabel
                        sx={{ marginBottom: (theme) => theme.spacing(3) }}
                        control={
                            <Checkbox
                                name="useGlobalTrigger"
                                checked={useGlobalTrigger}
                                onChange={(event) => {
                                    setUseGlobalTrigger(event.target.checked);
                                }}
                                color="primary"
                            />
                        }
                        label="Use Global Trigger"
                    />
                    {useGlobalTrigger &&
                        (availableGlobalTriggers.length === 0 ? (
                            <small>There are no global triggers available.</small>
                        ) : (
                            <ControlledSelect
                                className="DrawerFormField"
                                label="Global Trigger"
                                name="globalTriggerId"
                                values={availableGlobalTriggers}
                                formProps={props}
                                required
                            />
                        ))}
                </>
            )}
            <FormControlLabel
                sx={{ marginBottom: (theme) => theme.spacing(3) }}
                control={
                    <Checkbox
                        name="repeatable"
                        checked={values.minRepeatInterval > -1}
                        onChange={(event) => {
                            handleChange('minRepeatInterval', event.target.checked ? 0 : -1);
                        }}
                        color="primary"
                    />
                }
                label="Repeatable"
            />
            {values.minRepeatInterval > -1 && (
                <PositiveInRangeInput
                    name="minRepeatInterval"
                    label="Minimum time in seconds before rule can repeat"
                    value={values.minRepeatInterval}
                    setValue={(v: number | '') => handleChange('minRepeatInterval', v)}
                    validationError={props.errors.minRepeatInterval}
                    onBlur={props.handleBlur}
                    max="1200"
                    className="DrawerFormField"
                />
            )}
            <ControlledTextAreaInput
                name="comments"
                label="Comments"
                formProps={props}
                className="DrawerFormField"
            />
        </DrawerFormLayout>
    );
};

export default RuleForm;
