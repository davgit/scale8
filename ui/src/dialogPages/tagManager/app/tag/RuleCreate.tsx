import { FC } from 'react';
import RuleForm from '../../../../components/organisms/Forms/RuleForm';
import { FormProps, FormValidationResult } from '../../../../hooks/form/useFormValidation';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import CreateRuleQuery from '../../../../gql/mutations/CreateRuleQuery';
import nameValidator from '../../../../utils/validators/nameValidator';
import { CreateRuleResult } from '../../../../gql/generated/CreateRuleResult';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../../abstractions/DialogPreloadForm';
import { FetchAvailableGlobalTriggers } from '../../../../gql/generated/FetchAvailableGlobalTriggers';
import FetchAvailableGlobalTriggersQuery from '../../../../gql/queries/FetchAvailableGlobalTriggersQuery';

export type RuleValues = {
    name: string;
    globalTriggerId: string;
    minRepeatInterval: number;
    comments: string;
};

export type RuleFormProps = FormProps<RuleValues> & {
    availableGlobalTriggers?: { key: string; text: string }[];
};

const RuleCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const ruleCreateProps: DialogPreloadFormProps<
        FetchAvailableGlobalTriggers,
        RuleValues,
        RuleFormProps,
        CreateRuleResult
    > = {
        loadQuery: useQuery<FetchAvailableGlobalTriggers>(FetchAvailableGlobalTriggersQuery, {
            variables: { tagId: props.contextId },
        }),
        buildInitialStatePreload: () => ({
            name: '',
            globalTriggerId: '',
            comments: '',
            minRepeatInterval: -1,
        }),
        saveQuery: useMutation(CreateRuleQuery),
        mapSaveData: (ruleValues: RuleValues) => ({
            ruleCreateInput: {
                rule_group_id: props.id,
                name: ruleValues.name,
                min_repeat_interval:
                    ruleValues.minRepeatInterval > 0
                        ? ruleValues.minRepeatInterval * 1000
                        : ruleValues.minRepeatInterval,
                ...(ruleValues.globalTriggerId === ''
                    ? {}
                    : { global_trigger_id: ruleValues.globalTriggerId }),
                ...(ruleValues.comments === ''
                    ? {}
                    : {
                          comments: ruleValues.comments,
                      }),
            },
        }),
        buildFormProps: (
            formLoadedData: FetchAvailableGlobalTriggers,
            formValidationValues: FormValidationResult<RuleValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Create Rule',
            title: 'Create Rule',
            formInfoProps: buildStandardFormInfo('rules', 'Create'),
            handleDialogClose: props.handleDialogClose,
            availableGlobalTriggers: formLoadedData.getTag.revision.global_triggers.map((_) => ({
                key: _.id,
                text: _.name,
            })),
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createRule.id !== undefined,
        pageComponent: RuleForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Rule Group name too short',
            },
        ],
        ...props,
    };

    return (
        <DialogPreloadForm<
            FetchAvailableGlobalTriggers,
            RuleValues,
            RuleFormProps,
            CreateRuleResult
        >
            {...ruleCreateProps}
        />
    );
};

export default RuleCreate;
