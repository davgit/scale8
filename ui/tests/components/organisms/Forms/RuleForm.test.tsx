import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import RuleForm from '../../../../src/components/organisms/Forms/RuleForm';
import { RuleValues } from '../../../../src/dialogPages/tagManager/app/tag/RuleCreate';

const MockForm: FC = () => {
    const formProps = useMockFormProps<RuleValues>({
        name: '',
        globalTriggerId: '',
        minRepeatInterval: 0,
        comments: '',
    });

    return <RuleForm {...formProps} />;
};

describe('RuleForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
