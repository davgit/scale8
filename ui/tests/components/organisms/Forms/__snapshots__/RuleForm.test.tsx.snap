// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-1"
              id="mui-1-label"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-b4y4zd-MuiFormLabel-asterisk"
              >
                â€‰*
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-describedby="mui-1-helper-text"
                aria-invalid="true"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-1"
                name="name"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium Mui-focused Mui-required css-1sbojr7-MuiFormHelperText-root"
              id="mui-1-helper-text"
            >
              Required value
            </p>
          </div>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-18yztu3-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="repeatable"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Repeatable
            </span>
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-2"
              id="mui-2-label"
            >
              Minimum time in seconds before rule can repeat
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1v2pam7-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-2"
                max="1200"
                min="0"
                name="minRepeatInterval"
                type="number"
                value="0"
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-3"
              id="mui-3-label"
            >
              Comments
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1yxqjki-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-3"
                name="comments"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Comments
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
